backend/controller/ingress/request.go

```diff
@@ -122,6 +126,76 @@ func extractHTTPRequestBody(route *dal.IngressRoute, r *http.Request, ref *schem
 	return valueForData(bodyField.Type, bodyData)
 }
 
+// Takes the map of path parameters and transforms them into the appropriate type
+func manglePathParameters(params map[string]string, ref *schema.Ref, sch *schema.Schema) (any, error) {
+
+	paramsField, err := getField("pathParameters", ref, sch)
+	if err != nil {
+		return nil, err
+	}
+
+	switch paramsField.Type.(type) {
+	case *schema.Ref, *schema.Map:
+		ret := map[string]any{}
+		for k, v := range params {
+			ret[k] = v
+		}
+		return ret, nil
+	default:
+	}
+	// This is a scalar, there should only be a single param
+	// This is validated by the schema, we don't need extra validation here
+	for _, val := range params {
+
+		switch paramsField.Type.(type) {
+		case *schema.String:
+			return val, nil
+		case *schema.Int:
+			parsed, err := strconv.ParseInt(val, 10, 64)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse int from path parameter: %w", err)
+			}
+			return parsed, nil
+		case *schema.Float:
+			float, err := strconv.ParseFloat(val, 64)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse float from path parameter: %w", err)
+			}
+			return float, nil
+		case *schema.Bool:
+			// TODO: is anything else considered truthy?
+			return val == "true", nil
+		default:
+			return nil, fmt.Errorf("unsupported path parameter type %T", paramsField.Type)
+		}
+	}
+	// Empty map
+	return map[string]any{}, nil
+}
+
+// Takes the map of path parameters and transforms them into the appropriate type
+func mangleQueryParameters(params map[string]any, underlying map[string][]string, ref *schema.Ref, sch *schema.Schema) (any, error) {
+
+	paramsField, err := getField("query", ref, sch)
+	if err != nil {
+		return nil, err
+	}
+
+	if m, ok := paramsField.Type.(*schema.Map); ok {
+		if _, ok := m.Value.(*schema.Array); ok {
+			return params, nil
+		}
+	}
+	// We need to turn them into straight strings
+	newParams := map[string]any{}
+	for k, v := range underlying {
+		if len(v) > 0 {
+			newParams[k] = v[0]
```

Comment:
Potentially discarding entries with multiple values in the `[]string`. I don't have context here--can that happen?

Reply:
It could, but if you have chosen to map to a map[string]string your business logic likely can only handle a single value anyway. The vast majority of the time query parameters are singular.